File: rental-backend/src/main/java/com/realestate/rental/dto/MessageDto.java
package com.realestate.rental.dto;

import lombok.Data;

import java.time.LocalDateTime;
import java.util.UUID;

// MessageDto.java
@Data
public class MessageDto {
    private UUID id;
    private UUID chatRoomId;
    private UUID senderId;
    private String senderName;
    private String messageText;
    private Boolean isRead;
    private LocalDateTime createdAt;
}

File: rental-app-flutter/lib/features/chat/models/message_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'message_dto.freezed.dart';
part 'message_dto.g.dart';

@freezed
abstract class MessageDto with _$MessageDto {
  const factory MessageDto({
    required String id,
    required String chatRoomId,
    required String senderId,
    required String senderName,
    required String messageText,
    required bool isRead,
    required String createdAt,
  }) = _MessageDto;

  factory MessageDto.fromJson(Map<String, dynamic> json) => _$MessageDtoFromJson(json);
}
